src/creature.cpp

```diff
@@ -229,15 +229,19 @@ void Creature::onWalk()
 
 void Creature::onWalk(Direction& dir)
 {
-	if (hasCondition(CONDITION_DRUNK)) {
-		uint32_t r = uniform_random(0, 20);
-		if (r <= DIRECTION_DIAGONAL_MASK) {
-			if (r < DIRECTION_DIAGONAL_MASK) {
-				dir = static_cast<Direction>(r);
-			}
-			g_game.internalCreatureSay(this, TALKTYPE_MONSTER_SAY, "Hicks!", false);
-		}
+	uint32_t rand;
```

Comment:
why create this variable, if there is a case where it is not even used?

Reply:
At the end of the day, once it goes through the compiler, it will be translated into the same machine instructions.

For instance, the following two snippets:
```c
void foo() {
    int bar;

    if (rand() % 100 > 49) {
        return;
    }

    bar = 10;
    printf("%d", bar);

    return;
}
```
```c
void foo() {
    if (rand() % 100 > 49) {
        return;
    }

    int bar = 10;
    printf("%d", bar);

    return;
}
```
Are translated into the very same machine instructions
```asm
.LC0:
        .string "%d"
foo:
        sub     rsp, 8 ; allocate space in the stack for bar
        call    rand
        ; handle if
        cmp     eax, 49
        jle     .L5
        add     rsp, 8 ; deallocate b
        ret
.L5:
        ; handle b = 10 and printf
        add     rsp, 8 ; deallocate b
        jmp     printf
```

It is just a matter of style and giving "context" to a variable.

I have double checked the coding style guidelines and I have seen no entry for "variable locality" or so, but if you believe it will make the method more readable I can push the change to the branch.

Reply:
> I have double checked the coding style guidelines and I have seen no entry for "variable locality" or so

https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide#local-variables

Reply:
You just tilted me, not gonna lie. I appreciate the "effort" but your ---zero value--- comment clearly demonstrates you haven't thoroughly read the coding style guidelines. First of all, local variable does not equal variable locality. Secondly, because that very section lacks wording, the following snippet is completely correct under the TFS coding style guidelines:

```c
void foo() {
    int bar = 10;

    if (rand() % 100 > 49) {
        return;
    }

    printf("%d", bar);
    return;
}
```

Again, it does not solve the issue with the variable locality.